Astronomy/Model/FetchPhotoOperation.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  FetchPhotoOperation.swift
+//  Astronomy
+//
+//  Created by Rob Vance on 9/10/20.
+//  Copyright © 2020 Lambda School. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class FetchPhotoOperation: ConcurrentOperation {
+    var marsPhotoReference: MarsPhotoReference
+    var imageData: Data?
+    private var dataTask = URLSessionDataTask()
+    
+    init(marsPhotoReference: MarsPhotoReference) {
+        self.marsPhotoReference = marsPhotoReference
+    }
+    
+    override func start() {
+        state = .isExecuting
+        
+        let url = URL(string: "\(marsPhotoReference.imageURL.usingHTTPS!)")
+        let request = URLRequest(url: url!)
+        
+        dataTask = URLSession.shared.dataTask(with: request, completionHandler: { (data, _, error) in
+            if let error = error {
+                preconditionFailure("Error creating dataTask: \(error)")
+            }
+            
+            self.imageData = data
+            self.state = .isFinished
+        })
+        dataTask.resume()
+    }
+    override func cancel() {
+        dataTask.cancel()
```

Comment:
In this method besides calling cancel on your data task, you should also call super.cancel() and set your state to .isFinished.

> In the case of cancellation, it is still important to update the isFinished key path, even if the operation did not completely finish its task. Queued operations must report that they are finished before they can be removed from a queue.

https://developer.apple.com/documentation/foundation/nsoperation

``` swift
override func cancel() {
    dataTask.cancel()
    super.cancel()
    state = .isFinished
}
```